server/worker/macos_setup_assistant.go

```diff
@@ -309,8 +372,47 @@ func QueueMacosSetupAssistantJob(
 	}
 	job, err := QueueJob(ctx, ds, macosSetupAssistantJobName, args)
 	if err != nil {
-		return ctxerr.Wrap(ctx, err, "queueing job")
+		return 0, ctxerr.Wrap(ctx, err, "queueing job")
 	}
 	level.Debug(logger).Log("job_id", job.ID)
+	return job.ID, nil
+}
+
+func ProcessDEPCooldowns(ctx context.Context, ds fleet.Datastore, logger kitlog.Logger) error {
+	serialsByTeamId, err := ds.GetDEPAssignProfileExpiredCooldowns(ctx)
+	if err != nil {
+		return fmt.Errorf("getting cooldowns: %w", err)
+	}
+	if len(serialsByTeamId) == 0 {
+		logger.Log("msg", "no cooldowns to process")
+		return nil
+	}
+
+	// queue job for each team so that macOS setup assistant worker can pick it up and process it
+	for teamID, serials := range serialsByTeamId {
+		if len(serials) == 0 {
+			logger.Log("msg", "no cooldowns", "team_id", teamID)
+			continue
+		}
+		logger.Log("msg", "processing cooldowns", "team_id", teamID, "serials", fmt.Sprintf("%s", serials))
```

Comment:
nit, no need for the format call:

```suggestion
		logger.Log("msg", "processing cooldowns", "team_id", teamID, "serials", serials)
```